src/api/store/userprofile.py

```diff
@@ -132,11 +133,38 @@ def __init__(self):
     self.name = "UserProfile Store/DB"
   
   def validate_username(self, username):
-    if (not username):
-        return False
-
-    # checks if username already exists
-    return not UserProfile.objects.filter(preferred_name=username).exists()
+    # returns [is_valid, suggestion], error
+    try:    
+        if (not username):
+            return [False, None], None
+
+        # checks if username already exists
+        if not UserProfile.objects.filter(preferred_name=username).exists():
+            return [True, username], None
+
+        # username exists, finds next available closest username
+        usernames = list(UserProfile.objects.filter(preferred_name__istartswith=username).order_by('preferred_name').values_list("preferred_name", flat=True))
+
+        if len(usernames) == 1:
+            suggestion = username + "-0"
+            return [False, suggestion], None
+        numbers = [int(x[x.index("-")+1:]) for x in usernames[1:]]
```

Comment:
Would this expression be a problem if someone had a "-" in the middle of their name?

Reply:
Testing on Dev with the frontend-portal changes, the usernames returned when I tried to use BradHN were:
['BradHN', 'BradHN-0', 'BradHN24', 'BradHN5', 'BradHN-on-dev', 'BradHN-testing-!']